pkg/asset/installconfig/aws/validation.go

```diff
@@ -166,6 +166,122 @@ func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorLi
 	return field.ErrorList{field.Required(field.NewPath("platform", "aws", "amiID"), "AMI must be provided")}
 }
 
+// filterNotAvaibleEIPErrors checks if a given EIP allocation ID exists.
+func filterNotAvaibleEIPErrors(allErrs field.ErrorList) []error {
+	var invalidAndNotFoundEIPList []error
+	for _, err := range allErrs {
+		// Handle specific AWS error for non-existent allocation ID
+		if strings.Contains(err.Error(), "InvalidAllocationID.NotFound") {
+			invalidAndNotFoundEIPList = append(invalidAndNotFoundEIPList, err)
+		}
+	}
+	if len(invalidAndNotFoundEIPList) > 0 {
+		return invalidAndNotFoundEIPList
+	}
+	return nil
+}
+
+func validateEIPAllocations(ctx context.Context, meta *Metadata, fldPath *field.Path, config *types.InstallConfig) field.ErrorList {
+	if config.Platform.AWS.EIPAllocations == nil || len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return nil
+	}
+
+	var eipAllocations []awstypes.EIPAllocation
+	allErrs := field.ErrorList{}
+
+	if config.Platform.AWS != nil && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return nil
+	} else if len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) > 0 {
+		eipAllocations = config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer
+	}
```

Comment:
Why do you need this check? By this point `config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) > 0` or you would have returned in the previous check.

```suggestion
	eipAllocations = config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer
```